src/SFML/Audio/SoundFileWriterOpus.cpp

```diff
@@ -0,0 +1,259 @@
+// 1. The origin of this software must not be misrepresented;
+//    you must not claim that you wrote the original software.
+//    If you use this software in a product, an acknowledgment
+//    in the product documentation would be appreciated but is not required.
+//
+// 2. Altered source versions must be plainly marked as such,
+//    and must not be misrepresented as being the original software.
+//
+// 3. This notice may not be removed or altered from any source distribution.
+//
+////////////////////////////////////////////////////////////
+
+////////////////////////////////////////////////////////////
+// Headers
+////////////////////////////////////////////////////////////
+#include <SFML/Audio/SoundFileWriterOpus.hpp>
+#include <SFML/System/Err.hpp>
+#include <algorithm>
+#include <cstring>
+#include <ctime>
+#include <cassert>
+
+
+// Anonymous namespace
+namespace
+{
+    // Make sure to write int into buffer little endian
+    void writeUint32(unsigned char* buffer, sf::Uint32 value)
+    {
+        buffer[0] = static_cast<unsigned char>(value & 0x000000FF);
+        buffer[1] = static_cast<unsigned char>((value & 0x0000FF00) >> 8);
+        buffer[2] = static_cast<unsigned char>((value & 0x00FF0000) >> 16);
+        buffer[3] = static_cast<unsigned char>((value & 0xFF000000) >> 24);
+    }
+}
+
+namespace sf
+{
+namespace priv
+{
+////////////////////////////////////////////////////////////
+bool SoundFileWriterOpus::check(const std::string& filename)
+{
+    std::string extension = filename.substr(filename.find_last_of(".") + 1);
+    std::transform(extension.begin(), extension.end(), extension.begin(), ::tolower);
+
+    return extension == "opus";
+}
+
+
+////////////////////////////////////////////////////////////
+SoundFileWriterOpus::SoundFileWriterOpus() :
+m_channelCount (0),
+m_file         (),
+m_ogg          (),
+m_opus         (NULL),
+m_packageNumber(0)
+{
+}
+
+
+////////////////////////////////////////////////////////////
+SoundFileWriterOpus::~SoundFileWriterOpus()
+{
+    close();
+}
+
+
+////////////////////////////////////////////////////////////
+bool SoundFileWriterOpus::open(const std::string& filename, unsigned int sampleRate, unsigned int channelCount)
+{
+    // Save the channel count
+    m_channelCount = channelCount;
+    m_sampleRate = sampleRate;
+    std::srand(std::time(0));
+
+    // Initialize the ogg/opus stream
+    if (ogg_stream_init(&m_ogg, std::rand()) == -1)
+    {
+        err() << "Stream init of ogg/opus file \"" << filename << "\" failed" << std::endl;
+        close();
+        return false;
+    }
+    int status = 0;
+    m_opus = opus_encoder_create(sampleRate, channelCount, OPUS_APPLICATION_AUDIO, &status);
+    if (status != OPUS_OK)
+    {
+        err() << "Failed to write ogg/opus file \"" << filename << "\"" << std::endl;
+        if (status == OPUS_BAD_ARG)
+            err() << "Possible wrong sample rate, allowed are 8000, 12000, 16000, 24000, or 48000 Hz." << std::endl;
```

Comment:
@Bromeon Hm but it might be something different then the sample rate. The sample rate is only the most common issue here. E.g. the default sfml sample rate is 44,1kHz which is not supported, if really needed I could add a function to interpolate the sample rate to the next supported, but I think this would be overkill for sfml's needs, or?